tests/test_kem.c

```diff
@@ -127,6 +132,37 @@ static OQS_STATUS kem_test_correctness(const char *method_name) {
 		printf("shared secrets are equal\n");
 	}
 
+	if ((0 == strcasecmp(method_name, "ML-KEM-512")) || (0 == strcasecmp(method_name, "ML-KEM-768")) || (0 == strcasecmp(method_name, "ML-KEM-1024"))) {
+		// buffer to hold z and c. z is always 32 bytes
+		uint8_t *buff_z_c = NULL;
+		size_t length_z_c = (size_t)32 + kem->length_ciphertext;
+		buff_z_c = OQS_MEM_malloc(length_z_c) ;
+		if (NULL == buff_z_c) {
+			fprintf(stderr, "ERROR: OQS_MEM_malloc failed\n");
+			goto err;
+		}
+		// test rejection key by corrupting the secret key
+		secret_key[0] += 1;
+		uint8_t shared_secret_r[32]; // expected output
+		memcpy(buff_z_c, &secret_key[kem->length_secret_key - 32], 32);
+		memcpy(&buff_z_c[32], ciphertext, kem->length_ciphertext);
+		// calculate expected secret in case of corrupted cipher : shake256(z || c)
+		OQS_SHA3_shake256(shared_secret_r, 32, buff_z_c, length_z_c);
+		OQS_MEM_secure_free(buff_z_c, length_z_c);
+		rc = OQS_KEM_decaps(kem, shared_secret_d, ciphertext, secret_key);
+		if (rc != OQS_SUCCESS) {
+			fprintf(stderr, "ERROR: OQS_KEM_decaps failed for rejection testcase\n");
+			goto err;
+		}
+		rv = memcmp(shared_secret_d, shared_secret_r, kem->length_shared_secret);
```

Comment:
Did you check a constant-time test run with the code? I suspect the `shared_secret_r` variable should be declassified as the memcmp would lead to a false-positive. See the `OQS_TEST_CT_DECLASSIFY` statements earlier in the function.

Reply:
return value for decaps, shared_secret_r & shared_secret_d are declasified before usage